core/contracts/oracle/implementation/ContractCreator.sol

```diff
@@ -15,6 +16,14 @@ abstract contract ContractCreator {
         finderAddress = _finderAddress;
     }
 
+    function _requireWhitelistedCollateral(address collateralAddress) internal view {
```

Comment:
+1 I like where you placed the logical check for whether a currency is whitelisted. Any creator should check with the CollateralCurrencyWhitelist via this internal call.